plutus-pab-executables/test/full/Plutus/PAB/CoreSpec.hs

```diff
@@ -262,7 +262,8 @@ waitForTxOutStatusChangeTest = runScenario $ do
     void $ Simulator.payToPaymentPublicKeyHash w1 pk1 (Ada.toValue Ledger.minAdaTxOut)
     Simulator.waitNSlots 1
 
-  txOutStatus1'' <- Simulator.waitForTxOutStatusChange txOutRef1
+  let oldStatus = (TentativelyConfirmed 8 TxValid (Spent $ getCardanoTxId tx2))
+  txOutStatus1'' <- Simulator.waitForTxOutStatusChange' oldStatus txOutRef1
```

Comment:
My intuition is that it doesn't solve the hidden problem. Considering the fact the following testcase works
```
  assertEqual "tx output 1 should be tentatively confirmed of depth 1"
              (TentativelyConfirmed 1 TxValid (Spent $ getCardanoTxId tx2))
txOutStatus1'
```
and that a depth of > 8 means confirmed, and that we wait for at least 8 slots => this should mean that the transaction is definitely confirmed. Maybe there's a slot discrepancy between `waitForTxOutStatusChange` and `waitNSlots`.

Reply:
My hypothesis, based on the hydra error message, was that we don't wait enough and `waitForTxOutStatusChange` returns one of the status changes but not the latest. `TentativelyConfirmed 7 ~> TentativelyConfirmed 8` instead of `TentativelyConfirmed 8 ~> Confirmed`. That's why I tried to specify the status that we want to be changed from.

Now, I think, maybe we should instead wait for the specified status? Like `waitForTxOutStatusChangeTo status` and it will wait until the status will be changed to the given status. And we can provide a timeout 10s for example to throw an exception if the desired status wasn't reached. What do you think?

Reply:
I understand the solution you're proposing and it makes sense. However, I still feel like the current test should work. We're calling `Simulator.waitNSlots 1` 9 times, and a transaction should be confirmed once the transaction is 8 blocks deep. So it doesn't make sense for the test to fail unless there's a problem with the Simulator. I'll also take a look and see what the problem might be.

Reply:
I have added one more `waitNSlots 1` to `waitForTxOutStatusChangeTest` but `waitForTxStatusChangeTest` also failed with the similar error as on hydra. Maybe the tests in this test-suite are not isolated and they depend on each other? Maybe the simulator runs things and accumulates the delay?